### PR TITLE
checks: git: add PythonPEP517WithoutRevbump

### PR DESCRIPTION
**Pull request by thesamesam**
Fixes: https://github.com/pkgcore/pkgcheck/issues/498

TODO:
- [x] Add tests
- [x] Hoist up re.compile

### CODE REVIEW

File: src/pkgcheck/checks/git.py [hunk 1]
```diff
@@ -364,6 +378,19 @@ def modified_checks(self, pkgs, added):
         if pkg not in added and old_pkg.rdepend != new_pkg.rdepend:
             yield RdependChange(pkg=new_pkg)
 
+        # It's an existing package, but this doesn't fit into
+        # modified_checks() because we're working on the text.
+        python_pep517_regex = re.compile("^DISTUTILS_USE_PEP517=")
```
Comment (thesamesam) [inline src/pkgcheck/checks/git.py:383]: I'll move this out so we don't recompile it every time, obviously.

File: src/pkgcheck/checks/git.py [hunk 2]
```diff
@@ -364,6 +378,17 @@ def modified_checks(self, pkgs, added):
         if pkg not in added and old_pkg.rdepend != new_pkg.rdepend:
             yield RdependChange(pkg=new_pkg)
 
+        python_pep517_regex = re.compile("^DISTUTILS_USE_PEP517=")
+
+        def found_pep517_lines(cmp_pkg):
+            return any(python_pep517_regex.match(line) for line in cmp_pkg.ebuild.text_fileobj())
+
+        found_old_pep517_line = found_pep517_lines(old_pkg)
```
Comment (thesamesam) [inline src/pkgcheck/checks/git.py:386]: We can guard this with a distutils-r1 inherit check.

**Review by arthurzam — APPROVED** [review]

**Comment by mgorny** [comment]
On a semi-relevant note: I wonder if we should have something similar for EAPI bumps.

**Comment by thesamesam** [comment]
> On a semi-relevant note: I wonder if we should have something similar for EAPI bumps.

My view is that we should, for two reasons:
1. Risk of image changing (not just because of EAPI behaviour (also think of Portage keepdir w/ EAPI 8+, but also eclass functions may change)
2. Updating the EAPI in vdb so we can eventually remove old version support from Portage

**Review by arthurzam — APPROVED** [review]
Thank you!

**Comment by thesamesam** [comment]
Thanks!

**Pull request closed without merging**